### ISSUE TITLE
Rootfs Images not build for arm64 arch, hence Apple Silicon (M1) not working

### ISSUE THREAD

**Issue by carlonoelle** [mendix/docker-mendix-buildpack#157]
Hey all,

i tried to build using the current 3.5.2 buildpack, but got errors regarding the architecure:

```
#17 4.295 INFO: Selecting Mono Runtime: 5
#17 93.90 TYPE: 3
#17 93.90 V: AllocVector
#17 93.90 * Assertion at mini-amd64.c:217, condition `amd64_is_imm32 (disp)' not met
```

I fixed it for me by simply building the ubi8 and bionic images locally myself (on a macbook with m1 pro cpu) and using these ones for my build:

```
docker build \
  --build-arg BUILDPACK_XTRACE=true \
  --build-arg ROOTFS_IMAGE=<custom-ubi8-local>\
  --build-arg BUILDER_ROOTFS_IMAGE=<custom-bionic-local> \
  --tag testapp:1.0.0 .
```

Just wanted to let you know, maybe include the arm64 arch in your images published to dockerhub.

**Comment by danniarreza** [comment]
Hi, this is definitely something I've come across. How did you build the ubi8 / bionic images yourself?

**Comment by carlonoelle** [comment]
Hey @danniarreza ,

sorry for the late response. Still cannot build my app successfully, but didn't really had the time to investigate further.

I build the images using the dockerfiles inside this repo, since i build them on my machine the images are build for arm.

**Comment by tonytbn** [comment]
m1 m2 still can build as normal, just can not exec and run the image

**Comment by danniarreza** [comment]
@carlonoelle because of @tonytbn 's answer I was curious and managed to get it working (sort of). I used the default dockerfile config and used this command to build: 

docker buildx build -t <reponame>:<tag> --platform linux/amd64,linux/arm64/v8 --push .

Note that this requires our local docker on arm mac to have multi-platform build enabled (see https://docs.docker.com/build/building/multi-platform/). And afterwards, we can run this image on amd64-based linux environment (in my case, I tried it on a VPS, but running it on other amd64 linux should also work).
